packages/story-editor/src/app/api/apiProvider.js

```diff
@@ -80,6 +80,49 @@ function APIProvider({ children }) {
     withoutImages: [],
   });
 
+  // Add custom fields and prepare story response.
+  const prepareGetStoryResponse = (post) => {
+    const { _embedded: embedded = {}, _links: links = {} } = post;
+
+    post.author = {
+      id: embedded?.author?.[0].id || 0,
+      name: embedded?.author?.[0].name || '',
+    };
+
+    post.capabilities = {
+      hasPublishAction: Object.prototype.hasOwnProperty.call(
+        links,
+        'wp:action-publish'
+      ),
+      hasAssignAuthorAction: Object.prototype.hasOwnProperty.call(
+        links,
+        'wp:action-assign-author'
+      ),
+    };
+
+    post.lock_user = {
```

Comment:
I think we can easily use camelCase for all these vars. `lockUser`, `featuredMedia`, etc.

Reply:
I am thinking about the time when we document the expected response from these API callbacks. The fields should use the same pattern

https://github.com/google/web-stories-wp/blob/c0909f4495f80a12edb97f0ad8c30e8059ddac45/packages/story-editor/src/app/story/effects/useLoadStory.js#L47-L60

That's the same reason why and to make the API response look more platform agnotic, I have removed `_embeded` and `_links` because they look more WordPress specific. Thoughts?

Reply:
> I have removed `_embedded` and `_links` because they look more WordPress specific

Yeah and that makes total sense.

I was just suggesting using camelCase here so we don't do have to rename from `lock_user` to `lockUser` everywhere else. We could save this step by just using `lockUser` everywhere.

Reply:
> I was just suggesting using camelCase here

I think I could not understand. So if I use a camelCase for `lock_user` here inside [.then( transformGetStoryResponse )](https://github.com/google/web-stories-wp/blob/f311d45e15e9faa6d3e08e77067e92ec8ba38057/packages/wp-story-editor/src/api/story.js#L83) that would mean the expected API response fields from the callbacks would look something like this

```
          title: { raw: title },
          status,
          slug,
          date,
          modified,
          excerpt: { raw: excerpt },
          link,
          story_data,
          permalink_template,
          style_presets,
          password,
          preview_link,
          edit_link,
          embed_post_link,
          lockUser
```

Where `lockUser` is using camel case while the rest of the params are using `_`.

🤔

Reply:
It will surely save us one step but API we would document later would look something like this.

![image](https://user-images.githubusercontent.com/6297436/133304289-843ab58a-0185-4c82-acd1-e25c23940c2b.png)